<launch>
	
	<!-- arg name="filepath" default="/home/user/bboat_ws/src/bboat_pkg/mission_files/default_mission.txt" / -->
	<arg name="log" default="0" />

	<arg name="plot" default="0" />
	
	<arg name="filepath" default="/home/user/bboat_ws/src/bboat_pkg/mission_files/Guerledan_4.txt" />
	<arg name="obstacle_filepath" default="/home/user/bboat_ws/src/bboat_pkg/mission_files/egonav_obstacles.txt" />
	
	

	<arg name="mission_simu" default="0" />
	<arg name="mission_traj" default="1" />
	
	<arg name="record" default="false" />	
		



	<!-- Setting ROS control -->
	<node name="set_sysid"
	pkg="mavros"
	type="mavparam"
	args="set SYSID_MYGCS 1"
	output="screen"
	/>

	<!-- BBoat Traj Tracking node -->
	<node name="bboat_traj_tracking_node"
	type="traj_tracking_node.py"
	pkg="bboat_pkg"
	output="screen">
		<param name="filepath" value="$(arg filepath)" />
	</node>
	
	<!-- BBoat Driver node -->
	<node name="bboat_driver_node"
	type="driver_node.py"
	pkg="bboat_pkg"
	output="screen"
	/>

	<!-- BBoat Sensor node -->
	<node name="bboat_sensor_node"
	type="sensor_node.py"
	pkg="bboat_pkg"
	output="screen"
	/>

	<!-- BBoat Logger node -->
	<node name="bboat_logger_node"
	type="logger_node.py"
	pkg="bboat_pkg"
	output="screen">
		<param name="log" value="$(arg log)" />
		<param name="mission_traj" value="$(arg mission_traj)" />
	</node>

	<!-- BBoat Plotter node -->
	<node name="bboat_plotter_node"
	type="plotter_node.py"
	pkg="bboat_pkg"
	output="screen">
		<param name="plot" value="$(arg plot)" />
		<param name="mission_simu" value="$(arg mission_simu)" />
		<param name="mission_traj" value="$(arg mission_traj)" />
		<param name="obstacle_filepath" value="$(arg obstacle_filepath)" />
	</node>


	<!-- BBoat Controller node -->
	<node name="bboat_controller_node"
	type="controller_node.py"
	pkg="bboat_pkg"
	output="screen">
		<param name="mission_traj" value="$(arg mission_traj)" />
	</node>

	<!-- BBoat Camera node -->
	<!-- node name="bboat_camera_node"
	type="camera_node.py"
	pkg="bboat_pkg"
	output="screen"
	/ -->

	<!-- Joystick node -->
	<!-- node name="joystick_node"
	type="joy_node"
	pkg="joy"
	output="screen"
	/ -->

	<node pkg="rosbag" type="record" name="rosbag_record_cam"
       args="record -a"
       if="$(arg record)" />	
</launch>
